### ISSUE TITLE
`<taxonomy>` should be allowed as a child of `<category>`

### ISSUE THREAD

**Issue by TEITechnicalCouncil** [TEIC/TEI#496]
A taxonomy naturally contains categories:

~~~~~
<taxonomy xml:id="docType">
  <desc>Document type</desc>
  <category xml:id="dtManuscript">
    <catDesc>Handwritten manuscript</catDesc>
  </category>
  <category xml:id="dtPrint">
    <catDesc>Printed document</catDesc>
  </category>
</taxonomy>
~~~~~

and it's also obvious that categories should be able to nest:

~~~~~
  <category xml:id="dtManuscript">
    <catDesc>Handwritten manuscript</catDesc>
      <category xml:id="dtLetter">
        <catDesc>Handwritten letter</catDesc>
      </category>
      <category xml:id="dtMemo">
        <catDesc>Handwritten memo</catDesc>
      </category>
  </category>
~~~~~

It's clear that you could assign a document to any of the subcategories, and this would imply membership of the parent category; so if a document has `<catRef target="#dtLetter"/>`, then it is by definition also "dtManuscript".

However, there are some areas in a nested tree like this which don't follow this pattern. Consider this:

~~~~~
  <category xml:id="dtPaperSize">
    <catDesc>The size of paper on which a manuscript document is written.</catDesc>
    <category xml:id="dtA4">
      <catDesc>A4 paper</catDesc>
    </category>
    <category xml:id="dtA5">
      <catDesc>A5 paper</catDesc>
    </category>
  </category>
~~~~~

Any document may belong to either of the child categories (even both, if it happens to include both paper sizes); so we would see e.g. `<catRef target="#dtA4"/>`, meaning that the document falls into the category of documents which consist wholly or partially of A4 paper. However, this does not make any claim as to the parent category; it makes no sense to say that a letter "is" or "has" a paper size, without specifying what that size is.

In other words, the "category" of paper size is not a category at all; it's a taxonomy. This issue comes up frequently in complex, nested taxonomies. It would be valuable to be able to create a structure like this:

~~~~~
<category xml:id="dtManuscript">
  <catDesc>Handwritten manuscript</catDesc>
  <category xml:id="dtLetter">
    <catDesc>Handwritten letter</catDesc>
  </category>
  <category xml:id="dtMemo">
    <catDesc>Handwritten memo</catDesc>
  </category>
  <taxonomy xml:id="dtPaperSize">
    <desc>The size of paper on which a manuscript document is written.</desc>
    <category xml:id="dtA4">
      <catDesc>A4 paper</catDesc>
    </category>
    <category xml:id="dtA5">
      <catDesc>A5 paper</catDesc>
    </category>
  </taxonomy>
</category>
~~~~~

I submit that `<taxonomy>` should be available as a child of `<category>`, to allow for such rich multi-layered taxonomies.

Original comment by: @martindholmes

**Comment by martindholmes** [comment]
I've been a bit late getting to this, but: for the provision of nesting in `<taxonomy>` elements, the new content model would be:

~~~~~
<content>
    <choice xmlns="http://relaxng.org/ns/structure/1.0">
      <zeroOrMore>
        <choice>
          <ref name="model.glossLike"/>
          <ref name="model.descLike"/>
        </choice>
      </zeroOrMore>
      <oneOrMore>
        <ref name="category"/>
      </oneOrMore>
      <group>
        <group>
          <!-- this group a hack for odd2dtd processing -->
          <ref name="model.biblLike"/>
        </group>
        <zeroOrMore>
          <ref name="category"/>
        </zeroOrMore>
      </group>
    </choice>
  </content>
~~~~~
In other words, wherever `<category>` is allowed, `<taxonomy>` may also appear. I'll deal with the `<category>` content model in the next comment.

**Comment by martindholmes** [comment]
The change is essentially the same for the content model of `<category>`:

~~~~~
<content>
            <group xmlns="http://relaxng.org/ns/structure/1.0">
              <choice>
                <oneOrMore>
                  <ref name="catDesc"/>
                </oneOrMore>
                <zeroOrMore>
                  <choice>
                    <ref name="model.descLike"/>
                    <ref name="model.glossLike"/>
                  </choice>
                </zeroOrMore>
              </choice>
              <zeroOrMore>
                <choice>
                  <ref name="category"/>
                  <ref name="taxonomy"/>
                </choice>
              </zeroOrMore>
            </group>
          </content>
~~~~~

**Comment by martindholmes** [comment]
Finally, I would propose a small expansion of Chapter 2.3.7:

http://www.tei-c.org/release/doc/tei-p5-doc/en/html/HD.html#HD55

which would give a single small example of nesting two taxonomies inside a larger one (a simple concept), and one rather more detailed example showing how a taxonomy can be used nested inside a category as part of a larger structure.

**Comment by jamescummings** [comment]
My recollection was that council agreed this and my objections were outvoted. My personal opinion is that these should just be separate taxonomies next to each other rather than nested ones. To me a taxonomy is a single structured typology with nested hierarchies of categories inside that typology. To have nested taxonomies seems to be at odds with my understanding of it. Instead I would have sibling taxonomies. If the subjects to be classified are so typologically different as to need separate taxonomies then I don't think they belong in an overall taxonomy.  But that's just my two pence, I believe I was already outvoted.

**Comment by martindholmes** [comment]
I think you were outvoted, but if everyone else feels the same way, one solution would be to approach someone who knows a lot about taxonomies (Oyvind Eide, for instance) and ask if I'm completely mad or not.


I thought I should provide a clear statement of the new suggested content model before carrying it out.


Cheers,

Martin


Martin Holmes
UVic Humanities Computing and Media Centre
mholmes@uvic.ca
________________________________

**Comment by fabio-ciotti** [comment]
I must say I am a bit at odd with this solution. You end mixing different categorization. You are suggesting something that recalls faceted classification a la Ranganathan, but TEI taxonomy/cat model is in fact based on a a classical hierarchical classification scheme. I ask for a supplement of investigation, I can try to have some hints from experts I know.

**Comment by martindholmes** [comment]
Council decision 2015-10-28: The nesting of taxonomies is uncontroversial, so that should go ahead. The nesting of taxonomies inside categories is still making people uneasy, so MH will raise a new ticket for that specific issue, and will consult with ontology experts (FC and others) to get a sense of whether we have the right solution to the problem posed by this use-case. Green to go on nested taxonomies.

**Comment by martindholmes** [comment]
Changed content model of <taxonomy>` in commit ce068f3..efd406e. Examples and prose to come.

**Comment by martindholmes** [comment]
Added a remark in commit 7af34be..ad43f5c. Decided against adding another example, since there are lots on this spec, and the concept of nested taxonomies is not complex. If the companion ticket on nesting taxonomy inside category is approved, that would require examples. Closing this ticket.
